ci-operator/step-registry/cucushift/installer/check/gcp/user-labels/cucushift-installer-check-gcp-user-labels-commands.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [ ! -f "${SHARED_DIR}/user_tags_sa.json" ]; then
+  echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to find the key file of the IAM service-account for userTags testing on GCP."
+  exit 1
+fi
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+fi
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+INFRA_ID="$(oc get infrastructures.config.openshift.io cluster -o jsonpath='{.status.infrastructureName}')"
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${SHARED_DIR}/user_tags_sa.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+# User-defined labels validation. It will check if each user-defined label is applied. 
+# Return non-zero is one or more user-defined label absent. 
+# $1 - the current labels of the resource under question (JSON in compact format)
+function validate_user_labels() {
+  local -r current_labels_str="$1";  shift
+
+  if [ -z "${USER_LABELS}" ]; then
```

Comment:
I guess we can remove the pre check, after all, the check step, should only exists in a tag-label-enabled install workflow yaml file.